src/StateMaps/PDEConstrainedFunctionals.jl

```diff
@@ -531,6 +531,8 @@ function Fields.evaluate!(pcf::CustomPDEConstrainedFunctionals{N},φh) where N
   ignore_pullback = findall(!isnothing,vcat(analytic_dJ!, analytic_dC!))
   val, grad = val_and_jacobian(φ_to_jc, get_free_dof_values(φh);ignore_pullback)
 
+  @check length(val) == N + 1 "Expected $(N+1) constraints, φ_to_jc returned $(length(val))"

```

Comment:
"Expected N constraints, φ_to_jc returned (length(val)) which should match the number of constraints + 1"

Reply:
Ha, I was clearly not paying attention when I wrote this check 😆 Thanks!